File: client-java/instrumentation-shared/src/test/java/org/evomaster/client/java/instrumentation/shared/RegexUtilsTest.java
package org.evomaster.client.java.instrumentation.shared;

import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class RegexUtilsTest {


    @Disabled
    @Test
    public static void testParentheses(){

        String s = "foo";
        assertEquals("foo", RegexSharedUtils.removeParentheses(s));
        assertEquals("foo", RegexSharedUtils.removeParentheses("("+s+")"));
        assertEquals("foo", RegexSharedUtils.removeParentheses("(("+s+"))"));

        String x = "bar";
        assertEquals("(foo)(bar)", RegexSharedUtils.removeParentheses("("+s+")("+x+")"));
        assertEquals("(foo)(bar)", RegexSharedUtils.removeParentheses("(("+s+")("+x+"))"));
    }

}

File: core/src/test/kotlin/org/evomaster/core/parser/RegexUtilsTest.kt
package org.evomaster.core.parser

import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*

class RegexUtilsTest {

    @Test
    fun testIgnoreCaseRegex() {

        assertEquals("(a|A)(b|B)(c|C)\\Q123\\E(d|D)(e|E)(f|F)", RegexUtils.ignoreCaseRegex("aBc123DEf"))
    }

    @Test
    fun testIgnoreCaseControlChars() {

        assertEquals("(a|A)\\Q[](){}\\\"^\$.\\E(b|B)", RegexUtils.ignoreCaseRegex("a[](){}\\\"^\$.b"))
    }
}